FWCore/MessageLogger/interface/JobReport.h

```diff
@@ -269,6 +273,9 @@ namespace edm {
 
     ~JobReport();
 
+    // Insert information about the process
+    void reportProcess(std::string_view processName, ProcessConfigurationID reducedProcessID, ParameterSetID psetID);
```

Comment:
```suggestion
    void reportProcess(std::string_view processName, ProcessConfigurationID reducedProcessID, ParameterSetID const& psetID);
```
The `ProcessConfigurationID` is returned by value by the caller but the `ParameterSetID` is returned by `const &`. Given the ParameterSetID is a large object I'd suggest passing by reference into the function. I'm OK with the configuration id being by value or ref.

Reply:
Both `ProcessConfigurationID` and `ParameterSetID` are of type `edm::Hash<T>`
https://github.com/cms-sw/cmssw/blob/4eeb0e8b4a51b9df31ea74632d0a7e746151a1e7/DataFormats/Provenance/interface/ProcessConfigurationID.h#L8
https://github.com/cms-sw/cmssw/blob/4eeb0e8b4a51b9df31ea74632d0a7e746151a1e7/DataFormats/Provenance/interface/ParameterSetID.h#L21
that internally seems to hold `std::string`
https://github.com/cms-sw/cmssw/blob/4eeb0e8b4a51b9df31ea74632d0a7e746151a1e7/DataFormats/Provenance/interface/Hash.h#L107
https://github.com/cms-sw/cmssw/blob/4eeb0e8b4a51b9df31ea74632d0a7e746151a1e7/DataFormats/Provenance/interface/Hash.h#L45
https://github.com/cms-sw/cmssw/blob/4eeb0e8b4a51b9df31ea74632d0a7e746151a1e7/DataFormats/Provenance/interface/Hash.h#L30

so maybe this function should take both as `const&`.

Reply:
Both ID arguments are now taken as `const&`.